### ISSUE TITLE
ISE when trying to put an intersection type into the schema

### ISSUE THREAD

**Issue by msullivan** [edgedb/edgedb#2517]
```
CREATE TYPE Foo {create property y -> str};
CREATE TYPE Bar {create property y -> str};
CREATE TYPE Asdf { CREATE LINK uh := Foo[IS Bar] };
```
fails with a `NotImplementError` from `as_create_delta` in `TypeShell`.

It would probably be OK for now to not support this but we at least want a real error message.

**Comment by 1st1** [comment]
@elprans Do we want to fix this before 1.0? Or just fix this now by throwing a proper error?

**Comment by elprans** [comment]
Proper error for now
